src/main/java/uk/gov/companieshouse/reconciliation/function/compare_results/transformer/CompareCompanyStatusTransformer.java

```diff
@@ -7,23 +7,34 @@
 import org.apache.camel.Header;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Component;
+import uk.gov.companieshouse.reconciliation.function.compare_results.mapper.CompareCompanyNameResultMapper;
 import uk.gov.companieshouse.reconciliation.model.ResultModel;
 import uk.gov.companieshouse.reconciliation.model.Results;
 
 @Component
 public class CompareCompanyStatusTransformer {
+
+    private final CompareFieldsResultsTransformer transformer;
+    private final CompareCompanyNameResultMapper mapper;
+
     @Autowired
-    private CompareFieldsResultsTransformer transformer;
+    public CompareCompanyStatusTransformer(CompareFieldsResultsTransformer transformer, CompareCompanyNameResultMapper mapper) {
+        this.transformer = transformer;
+        this.mapper = mapper;
+    }
 
     public List<Map<String, Object>> transform(@Header("SrcList") Results srcResults,
             @Header("SrcDescription") String srcDescription,
             @Header("TargetList") Results targetResults,
             @Header("TargetDescription") String targetDescription,
             @Header("RecordKey") String recordKey) {
-        return transformer.transform(srcResults, srcDescription, targetResults, targetDescription, recordKey, this::generateMappings);
+        return transformer.transform(srcResults, srcDescription, targetResults, targetDescription, recordKey, mapper::generateMappings);
```

Comment:
We may want to perform other mapping operations on a CompanyResultsMappable; let's change the signature of `CompareFieldsResultsTransformer.transform()` so that it accepts a ResultsMappable object.

Reply:
Resolved in latest commit.